arangod/Agency/State.cpp

```diff
@@ -1002,16 +1002,24 @@ index_t State::loadCompacted() {
 
 /// Load persisted configuration
 bool State::loadOrPersistConfiguration() {
-  std::string const aql(
-      "FOR c in configuration FILTER c._key==\"0\" RETURN c.cfg");
+  // this must be in a lambda so the query lifetime does not overlap
```

Comment:
well lambda is not required it's possible to do same without it.
we can just write query.reset();

But I think it doesn't matter